packages/editor/src/Editor.tsx

```diff
@@ -871,6 +875,36 @@ const EditorContent = ({
     if (window.editorWindowAPI.onUpdate) {
       window.editorWindowAPI.onUpdate();
     }
+
+    const shopstoryCanvasIframe = window.document.getElementById(
+      "shopstory-canvas"
+    ) as HTMLIFrameElement | undefined;
+
+    if (meta && renderableContent && editorContext && externalData) {
+      shopstoryCanvasIframe?.contentWindow?.postMessage(
+        {
+          type: "@easyblocks/init",
+          meta: JSON.stringify(meta),
+          compiled: JSON.stringify(renderableContent),
+          editorContext: JSON.stringify({
```

Comment:
The problem with `postMessage` is that it requires you to create serialized copies. In that case `editorContext` is well, a context, and there should be only one reference of it. If you use a copy of editor context in the canvas then it just can't work properly.